packages/forms/src/UIForm/Widget/condition.js

```diff
@@ -1,18 +1,167 @@
 import get from 'lodash/get';
-import includes from 'lodash/includes';
+import findIndex from 'lodash/findIndex';
 
-function getValue(value, strategy) {
-	if (!strategy) {
+/**
+ * Tries to ensure the returned value is a number.
+ * It supports string and number for now.
+ */
+function stringOrNumberToNumber(value) {
+	if (typeof value === 'number') {
 		return value;
 	}
-	switch (strategy.toLowerCase()) {
-		case 'length':
-			if (value && value.length) {
-				return value.length;
+	if (typeof value === 'string') {
+		return Number(value);
+	}
+	const error = { error: 'the passed value is not a string or a number', value };
+	throw error;
+}
+
+/**
+ * Tries to make the returned value a string.
+ * It supports number and string as input.
+ * In case of an error it will call onError. By default it throws an exception.
+ */
+function stringOrNumberToString(
+	value,
+	onError = val => {
+		const error = { error: 'the passed value is not a string or a number', value: val };
+		throw error;
+	},
+) {
+	if (!value) {
+		return value;
+	}
+	if (typeof value === 'number') {
+		return JSON.stringify(value);
+	}
+	if (typeof value === 'string') {
+		return value.toString();
+	}
+	return onError(value);
+}
+
+/**
+ * "foo=bar;dummy=1" will return [{foo:"bar",dummy:"1"}]
+ */
+function parseParameters(parameterString) {
+	return parameterString
+		.split(';')
+		.map(it => {
+			const sep = it.indexOf('=');
+			if (sep > 0) {
+				return {
+					[it.substring(0, sep).trim()]: it.substring(sep + 1, it.length).trim(),
+				};
 			}
-			return 0;
+			return {
+				value: it.trim(),
+			};
+		})
+		.reduce(
+			(a, v) => ({
+				...a,
+				...v,
+			}),
+			{},
+		);
+}
+
+/**
+ * parses a strategy based on this pattern "name(parameters)" and returns
+ * an object containing { name, params }.
+ */
+function parseStrategy(strategy) {
+	if (!strategy) {
+		return strategy;
+	}
+
+	const start = strategy.indexOf('(');
+	if (start > 0) {
```

Comment:
Parsing a string is entirely the job of RegExp, one that could do the job :
`^([a-zA-Z]{1}\w*)(?:\((.+)?\))?$`
Will extract the name as second match and parameters as third match (or undefined, if no parentheses or empty parentheses)

No need to deal with indexes this way and lot's of checks. There will just remain the need to check if there is parameters against undefined (to verify with lots of units tests).

Reply:
using indices is faster (keep in mind it is called in render) and is less fragile than your regex which doesnt guarantee the parenthesis symmetry I think :(.

Reply:
There is unit tests for robustness.
If the values doesn't change, they must be some caching with memoization somehow somewhere to avoid recomputation. And if the values change often, there is other questions to asks, like why are we changing the strategy really often ?.

I never find the fast argument relevant on frontend for strict computation only. It's more often (always?) some i/o latency, some calls to compute values already computed, some computations not needed at the current time (like a whole list not actually rendered) which can be deferred ; which cause the real bad performances, aside with bad architecture.
I have found a lot of cases, where fast is the first thing searched, resulting in cumbersome code in the big picture, which result in others developers not understanding what's actually done, making others changes on top without refactoring because they cannot change what they don't understand, resulting at last in a poor final architecture app and so bad performances :p

I don't understand the thing with the parenthesis, what's need to be guarantee ?

Reply:
ps: dealing with indexes is as fragile as regex ^^'

Reply:
We can memoize it but it is not really needed with current implementation I think (which is always better since memoization is a workaround due to the lack of listeners :().

If you have "foo(" you must not parse parameters, your regex will let it do it if I read it right.
The other advantage i see using this implementation (and I will agree it is not an advantage for you but it is for me and talend ecosystem ;)) is that it is 1-1 the implementation we have in talend studio and since this is a feature we must offer on both sides it is simpler to align it.
Depending how we make this trivial language evolving, regex can kill us very quickly whereas this parsing will always work and be reliable cause it enables any syntax whereas regex are limited from my past experiences.